### ISSUE TITLE
How to extract original model's weight from trained QAT model?

### ISSUE THREAD

**Issue by doomooo** [tensorflow/model-optimization#977]
System information

TensorFlow version (you are using): 2.8
Are you willing to contribute it (Yes/No):
Motivation

How to extract the original model's weight from the trained QAT model?
I want to extract the original model's weight and QAT params, so I can set int8 params in TensorRT manually.

**Comment by doomooo** [comment]
Hi, I wonder how to extract the original model's weight from the trained QAT model? @thaink

**Comment by doomooo** [comment]
Anyone can help take a look? Thanks! @thaink @rino20

**Comment by thaink** [comment]
Isn't the original model weights in checkpoint files or variables/ directory?

**Comment by doomooo** [comment]
Thanks for your reply!  @thaink 
No, the weight is in quant_warp, and the name is changed.

**Comment by doomooo** [comment]
I can extract the weights with some codes or tricks,  but I wonder if there exist official methods. @thaink

**Comment by thaink** [comment]
I don't think we have an official way for that. This use with TensorRT isn't in our set of use-cases for now.

**Comment by inho9606** [comment]
@Xhark could you review this thread?
Thanks

**Comment by WillLiGitHub** [comment]
model: original model
        model_q: quantized model

        for ly in model.layers:
            name = 'quant_' + ly.name
            if len(ly.variables) > 0:
                lly = model_q.get_layer(name)
                for var in ly.variables:
                    for vvar in lly.variables:
                        if var.name == vvar.name:
                            var.assign(tf.Variable(vvar.numpy()))
                            break

**Comment by doomooo** [comment]
@WillLiGitHub Thanks！ I have extracted weights in this way.
